.github/actions/setup-cached-mamba/action.yml

```diff
@@ -0,0 +1,62 @@
+# Description of this workflow
+
+name: "Setup cached mamba"
+inputs:
+  python-version:
+    required: true
+    type: string
+  env-prefix:
+    required: true
+    type: string
+  env-label:
+    required: true
+    type: string
+  additional-env-file:
+    required: false
+    type: string
+  run-pyironconfig:
+    required: false
+    type: boolean
+    default: false
+runs:
+  using: "composite"
+  steps:
+  - uses: actions/checkout@v2
+  - name: Write environment
+    shell: bash -l {0}
+    run: |
+      if [ "${{ inputs.additional-env-file }}" == "" ]; then
+        cp .ci_support/environment.yml environment.yml
+      else
+        python .ci_support/condamerge.py --base .ci_support/environment.yml --add ${{ inputs.additional-env-file }} > environment.yml
+      fi
+  - name: Setup Mambaforge
+    uses: conda-incubator/setup-miniconda@v2
+    with:
+      python-version: ${{ inputs.python-version }}
+      miniforge-variant: Mambaforge
+      channels: conda-forge
+      channel-priority: strict
+      activate-environment: my-env
+      use-mamba: true
+  - name: Set cache date and number
+    shell: bash -l {0}
+    run: |
+      echo "DATE=$(date +'%Y%m%d')" >> $GITHUB_ENV
+      cat .github/variables/cache_number.env >> $GITHUB_ENV
+  - uses: actions/cache@v2
+    with:
+      path: ${{ inputs.env-prefix }}
+      key: ${{ inputs.env-label }}-conda-${{ hashFiles('environment.yml') }}-${{ env.DATE }}-${{ env.CACHE_NUMBER }}
+    id: cache
+  - name: Update environment
+    shell: bash -l {0}
+    run: mamba env update -n my-env -f environment.yml
+    if: steps.cache.outputs.cache-hit != 'true'
+  - name: Run pyironconfig
+    shell: bash -l {0}
+    run: python .ci_support/pyironconfig.py
+    if: ${{ inputs.run-pyironconfig }}
+  - name: Setup
+    shell: bash -l {0}
+    run: pip install --no-deps .
```

Comment:
These lines are not responsible to build the environment itself, but to configure pyiron, as you know. They sometimes change between the workflows, e.g. to install other stuff first and then installing `.` (e.g. in compatibility workflows checking out and using another repo).

Reply:
My understanding is that `pip install --no-deps` is responsible for building the part of the environment that is the code in the repository itself 😜 But yeah, the other bit is definitely pyiron config and not python env config.

In an earlier draft I actually had left these two parts out of the extracted action. For this PR I'm comfortable leaving them in, since the single input arg and if call mean that I can handle *all* my CI needs from this action. For more generic behaviour I am on board with re-thinking this. Maybe we could have the Mamba setup as it is here, and then have a second action with a bunch of boolean flags for doing stuff typically done in these last two steps? Nothing and `pyironconfig.py` are definitely a very common actions here, but maybe we can abstract out compatibility work as well?